### PR TITLE
Semver changes

### PR DESCRIPTION
**Pull request by Roms1383**
One of my user requested me to expose plugin version as early as possible, so I made a couple of non-breaking changes:
- ~~add different constructors for `alpha`, `beta` and `rc`~~
- ~~add `impl Display for Semver` to be able to return it conveniently as a string~~

I know semver can be a wide topic / opinionated but I had to settle down on a naming convention, so I tried to be as standard as possible.

*update: see my comment below, I simplified its implementation.*

### CODE REVIEW

**Comment by Roms1383** [comment]
The idea being to then use it like e.g.
```rs
GlobalExport(global!(c"MyPlugin.Version", version)),
GlobalExport(global!(c"MyPlugin.SemanticVersion", semantic_version)),

fn version() -> String { format!("{}", MyPlugin::VERSION) }
fn semantic_version() -> Vec<u32> { [u32; 5]::from(MyPlugin::VERSION).to_vec() }
```
and
```swift
module MyPlugin

public native func Version() -> String;
public native func SemanticVersion() -> array<Uint32>;
```

**Comment by Roms1383** [comment]
I chose to simply allow specifying the complete `SemVer` and `impl IntoRepr` for it so that crate consumers are not forced into a particular naming convention and are free to interpret the version meaning as they see fit.

**Comment by Roms1383** [comment]
That changes the above example like so:
```rs
GlobalExport(global!(c"MyPlugin.Version", version)),
fn version() -> RedArray<u32> { MyPlugin::VERSION }
```
and 
```swift
module MyPlugin

public native func Version() -> array<Uint32>;
```

**Review by jac3km4 — CHANGES_REQUESTED** [review]
I think SemVer's redscript repr should be `[Uint16; 3]` instead of `[u32]`, the pre-release parts are non-standard so IMO should not be exposed, you can use `StaticArray<u16, 3>` to expose it as a static array (it might need some impls added)

**Comment by Roms1383** [comment]
Ok so I added `StaticArray<u16, 3>` as the default impl for `IntoRepr`, but i left myself the opportunity to turn it `Into` a `StaticArray<u16, 5>` since i tend to use `alpha`, `beta`, `rc` actively and it's very convenient for that purpose.

File: build.rs [hunk 1]
```diff
@@ -45,4 +45,23 @@ fn main() {
...
+        sizes.push('"');
+        sizes.push_str(i.to_string().as_str());
+        sizes.push('"');
+        sizes.push(',');
+    }
+    sizes.push_str("\n      _ => unimplemented!()");
+    sizes.push_str("\n    }");
+    sizes.push('\n');
+    sizes.push('}');
+    std::fs::write(out_path.join("sizes.rs"), sizes).expect("Couldn't write const_digit_str!");
```
Comment (jac3km4) [inline build.rs:66]: I don't like the idea of generating the function from here, I can rewrite it as a const fn later, for now let's just hardcode a const fn that matches on numbers from 0 to 8.

Comment (Roms1383) [reply build.rs]: Reverted in f4aa009.

**Pull request merged**